docs/guides/request_storage.mdx

```diff
@@ -0,0 +1,113 @@
+---
+id: request-storage
+title: Request storage
+description: How to store the requests your crawler will go through
+---
+
+import ApiLink from '@site/src/components/ApiLink';
+
+import Tabs from '@theme/Tabs';
+import TabItem from '@theme/TabItem';
+import CodeBlock from '@theme/CodeBlock';
+
+import RqBasicExample from '!!raw-loader!./code/request_storage_rq_basic.py';
+import RqWithCrawlerExample from '!!raw-loader!./code/request_storage_rq_with_crawler.py';
+import RqWithCrawlerExplicitExample from '!!raw-loader!./code/request_storage_rq_with_crawler_explicit.py';
+
+import RlBasicExample from '!!raw-loader!./code/request_storage_rl_basic.py';
+import RlWithCrawlerExample from '!!raw-loader!./code/request_storage_rl_with_crawler.py';
+
+import RsDoNotPurgeExample from '!!raw-loader!./code/request_storage_do_not_purge.py';
+import RsPurgeExplicitlyExample from '!!raw-loader!./code/request_storage_purge_explicitly.py';
+
+This guide explains the different types of request storage available in Crawlee, how to store the requests that your crawler will process, and which storage type to choose based on your needs.
+
+## Request providers overview
+
+All request storage types in Crawlee implement the same interface - <ApiLink to="class/RequestProvider">`RequestProvider`</ApiLink>. This unified interface allows them to be used in a consistent manner, regardless of the storage backend. The request providers are managed by storage clients - subclasses of <ApiLink to="class/BaseStorageClient">`BaseStorageClient`</ApiLink>. For instance, <ApiLink to="class/MemoryStorageClient">`MemoryStorageClient`</ApiLink> stores data in memory while also can offload them to the local directory. Data are stored in the following directory structure:
```

Comment:
```suggestion
All request storage types in Crawlee implement the same interface - <ApiLink to="class/RequestProvider">`RequestProvider`</ApiLink>. This unified interface allows them to be used in a consistent manner, regardless of the storage backend. The request providers are managed by storage clients - subclasses of <ApiLink to="class/BaseStorageClient">`BaseStorageClient`</ApiLink>. For instance, <ApiLink to="class/MemoryStorageClient">`MemoryStorageClient`</ApiLink> stores data in memory while it can also offload them to the local directory. Data are stored in the following directory structure:
```